proto/data_objects.proto

```diff
@@ -0,0 +1,268 @@
+  string version = 5;
+
+  //REQUIRED, the bundle SHA256 checksum which is a SHA256 checksum of all the
+  // SHA256 checksums of the files that make up this bundle
+  string checksum =  6;
+
+  //OPTIONAL
+  string dataset_id = 7;
+
+  //OPTIONAL
+  repeated string urls = 8;
+
+  //OPTIONAL
+  string description = 9;
+
+  //OPTIONAL
+  repeated string aliases = 10;
+
+  //OPTIONAL
+  google.protobuf.Struct info = 12;
+
+  // OPTIONAL the id of the file as submitted
+  string submitted_id = 14;
+
+}
+
+message DataObjectCloudsResponse {
+  repeated string clouds = 1;
+}
+
+message DataObjectSignedURLResponse {
+  string url = 1;
+}
+
+message GetDataObjectCloudRequest {
+  string id = 1;
+  string cloud = 2;
+}
+
+message DataObjectNativeURLResponse {
+  string url = 1;
+  string key = 2;
+  string secret_key = 3;
+}
+
+message AddDataBundleObjectResponse {
+  string id = 1;
+}
+
+message GetDataBundleObjectRequest {
+  string id = 1;
+}
+
+message GetDataBundleObjectChecksumRequest {
+  string checksum = 1;
+}
+
+message DeleteDataBundleObjectRequest {
+  string id = 1;
+}
+
+message DeleteDataObjectRequest {
+  string id = 1;
+}
+
+message DataBundleStatusResponse {
+  //REQUIRED
+  string id = 1;
+
+  //REQUIRED
+  repeated DataBundleObject data_bundle_versions = 2;
+}
+
+message DataBundleChecksumStatusResponse {
+  //REQUIRED
+  string checksum = 1;
+
+  //REQUIRED
+  repeated DataBundleObject data_bundle_versions = 2;
+}
+
+message ListDataBundleObjectsRequest {
+  //OPTIONAL
+  string id = 1;
+
+  //OPTIONAL
+  string dataset_id = 3;
+
+  //OPTIONAL
+  string submitted_id = 4;
+
+  //OPTIONAL
+  // Specifies the maximum number of results to return in a single page.
+  // If unspecified, a system default will be used.
+  int32 page_size = 5;
+
+  //OPTIONAL
+  // The continuation token, which is used to page through large result sets.
+  // To get the next page of results, set this parameter to the value of
+  // `next_page_token` from the previous response.
+  string page_token = 6;
+
+}
+
+message ListDataBundleObjectsResponse {
+  // The list of datasets.
+  repeated DataBundleObject data_bundle_objects = 1;
+
+  // The continuation token, which is used to page through large result sets.
+  // Provide this value in a subsequent request to return the next page of
+  // results. This field will be empty if there aren't any additional results.
+  string next_page_token = 2;
+}
+
+message FileStatusResponse {
+  //REQUIRED
+  string id = 1;
+
+  //REQUIRED
+  repeated DataObject file_versions = 2;
+
+}
+
+message GetDataObjectChecksumResponse {
+  //REQUIRED
+  string checksum = 1;
+
+  //REQUIRED
+  repeated DataObject file_versions = 2;
+
+}
+
+///-------------------------------------------------------------------
+// Resource: a file, API or other resource
+///-------------------------------------------------------------------
+message DataObject {
+  //REQUIRED
+  string id = 1;
+
+  //REQUIRED
+  string file_name = 2;
```

Comment:
can this be optional? data object doesn't necessarily need a name, or doesn't need to store the name in the dataobject

Reply:
I advise removing filename and using Alias for everything.

Downloading files by filename is risky if two people name different files similarly. They will overwrite files unintentionally.

Reply:
removing filename will be great

Reply:
+1 remove file name